File: Modules/SEDO/EMS/SIE.Web.EMS.EarlierStage/Projects/Commands/DeleteProjectCommand.js
SIE.defineCommand('SIE.Web.EMS.EarlierStage.Projects.Commands.DeleteProjectCommand', {
    extend: 'SIE.cmd.Delete',
    meta: { text: "删除", group: "edit", iconCls: "icon-DeleteEntity icon-red" },
    canExecute: function (view) {
        var selectModels = view.getSelection();
        if (selectModels.length == 0) return false;
        var res = true;
        SIE.each(selectModels, function (model) {
            if (model.data.ApprovalStatus !== 10) {
                res = false;
                return false;
            }
        });
        return res;
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS.EarlierStage/Projects/Commands/DeleteProjectCommand.cs
using SIE.Domain;
using SIE.EMS.EarlierStage.Projects;
using SIE.Web.Command;
using System.Collections.Generic;
using System.Linq;

namespace SIE.Web.EMS.EarlierStage.Projects.Commands
{
    /// <summary>
    /// 删除项目
    /// </summary>
    [JsCommand("SIE.Web.EMS.EarlierStage.Projects.Commands.DeleteProjectCommand")]
    public class DeleteProjectCommand : DeleteCommand
    {
        /// <summary>
        /// 保存前动作
        /// </summary>
        /// <param name="data"></param>
        protected override void OnSaving(EntityList data)
        {
            base.OnSaving(data);
            var ids = new List<double>();
            data.DeletedList.ForEach(p =>
            {
                var model = p as Project;
                if (model != null)
                    ids.Add(model.Id);
            });
            RT.Service.Resolve<ProjectController>().DeleteCheckProjectState(ids);
        }
    }
}
